Source/Core/DolphinQt/Config/GeckoCodeWidget.cpp

```diff
@@ -222,11 +231,46 @@ void GeckoCodeWidget::SaveCodes()
 {
   IniFile game_ini_local;
   game_ini_local.Load(File::GetUserPath(D_GAMESETTINGS_IDX) + m_game_id + ".ini");
+
   Gecko::SaveCodes(game_ini_local, m_gecko_codes);
 
   game_ini_local.Save(File::GetUserPath(D_GAMESETTINGS_IDX) + m_game_id + ".ini");
 }
 
+void GeckoCodeWidget::OnContextMenuRequested()
+{
+  QMenu menu;
+
+  menu.addAction(tr("Sort Alphabetically"), this, &GeckoCodeWidget::SortAlphabetically);
+
+  menu.exec(QCursor::pos());
+}
+
+void GeckoCodeWidget::SortAlphabetically()
+{
+  m_code_list->sortItems();
+  OnListReordered();
+}
+
+void GeckoCodeWidget::OnListReordered()
+{
+  // Reorder codes based on the indices of table item
+  std::vector<Gecko::GeckoCode> codes;
+  codes.reserve(m_gecko_codes.size());
+
+  for (int i = 0; i < m_code_list->count(); i++)
+  {
+    const int index = m_code_list->item(i)->data(Qt::UserRole).toInt();
+
+    codes.push_back(m_gecko_codes[index]);
```

Comment:
Same here. (in the context of whatever containers a `Gecko::GeckoCode` contains)